evalml/pipelines/time_series_pipeline_base.py

```diff
@@ -80,20 +80,37 @@ def _move_index_forward(index, gap):
         else:
             return index + gap
 
-    @staticmethod
-    def _are_datasets_separated_by_gap(train_index, test_index, gap):
-        """Determine if the train and test datasets are separated by gap number of units.
+    def _are_datasets_separated_by_gap_time_index(self, train, test, gap):
+        """Determine if the train and test datasets are separated by gap number of units using the time_index.
 
         This will be true when users are predicting on unseen data but not during cross
         validation since the target is known.
         """
         gap_difference = gap + 1
-        index_difference = test_index[0] - train_index[-1]
-        if isinstance(
-            train_index, (pd.DatetimeIndex, pd.PeriodIndex, pd.TimedeltaIndex)
-        ):
-            gap_difference *= test_index.freq
-        return index_difference == gap_difference
+
+        train_copy = train.copy()
+        test_copy = test.copy()
+        train_copy.ww.init(time_index=self.time_index)
+        test_copy.ww.init(time_index=self.time_index)
+
+        X_frequency_dict = train_copy.ww.infer_temporal_frequencies(
+            temporal_columns=[train_copy.ww.time_index]
+        )
+        freq = X_frequency_dict[test_copy.ww.time_index]
+        if freq is None:
```

Comment:
This was added regarding a conversation with @freddyaboulton. He raised the valid point that we don't want AutoMLSearch to pass only to fail when a user tries and uses `predict`. In this case, AutoMLSearch will raise this error if the `time_index` does not have values that can be inferred.
I'm up for a discussion regarding this since I know it touches a lot of time series and is essentially the first time that our whole algorithm will be directly exposed to a proper interval frequency check on time series (outside of DateTimeFormatDataCheck).